docs/semgrep-ci/running-semgrep-ci-with-semgrep-app.md

```diff
@@ -328,11 +329,107 @@ SEMGREP_PR_ID="44"
+1. In the Bitbucket workspace, click the cogwheel icon next to your profile icon in the top right.
+1. Click **Workspace settings** > **Access tokens**.
+1. Click **Create Workspace Access Token**.
+1. Create a new workspace variable by entering **SEMGREP_APP_TOKEN** into the **Name** field, and then paste the value of the token in the **Value** field.
+1. Enable the **Secured** option, and then click **Add**.
+
+###### Creating and adding a repository access token
+
+Create a repository access token in Bitbucket. Follow the instructions in [Create a repository Access Token](https://support.atlassian.com/bitbucket-cloud/docs/create-a-repository-access-token/). Alternatively, follow these steps to create a repository access token:
+
+1. In Bitbucket, select the repository where you want Semgrep to run in CI.
+1. Click **Repository settings** > **Access tokens**.
+1. Click **Create Repository Access Token**.
+1. Create a token name. Note: As Semgrep comments to your PRs, use **Semgrep** as the name for this token.
+1. Under **Scopes**, select the **Read**, and **Write** permissions for pull requests.
+1. Click **Create**.
+1. Copy the value of the repository access token, and then click **Close**.
+1. Click **Repository variables**.
+1. Create a new workspace variable by entering **PAT** into the **Name** field, and then paste the workspace access token in the **Value** field.
+1. Enable the **Secured** option, and then click **Add**.
+
+Create and add a `SEMGREP_APP_TOKEN` to establish the communication between your repository and the Semgrep Cloud Platform by following these steps:
+
+1. Create `SEMGREP_APP_TOKEN` by following the [Creating a `SEMGREP_APP_TOKEN`](/semgrep-ci/running-semgrep-ci-with-semgrep-app/#creating-a-semgrep_app_token) documentation. Copy the value of the token created in the Semgrep Cloud Platform.
+1. In Bitbucket, select the repository where you want Semgrep to run in CI.
+1. Click **Repository settings** > **Access tokens**.
+1. Click **Create Workspace Access Token**.
+1. Create a new workspace variable by entering **SEMGREP_APP_TOKEN** into the **Name** field, and then paste the value of the token in the **Value** field.
+1. Enable the **Secured** option, and then click **Add**.
+
+##### Adding Semgrep to your Bitbucket CI pipeline for PR comments
+
+To receive comments on PRs, add the following code to your `bitbucket-pipelines.yml` file:
+
+```yaml
+image: atlassian/default-image:latest
+
+pipelines:
+  pull-requests:
+    '**':
+      - step:
+          name: 'Run Semgrep scan in PR branch'
+          image: returntocorp/semgrep
+          script:
+            - export SEMGREP_APP_TOKEN=$SEMGREP_APP_TOKEN
+            - export SEMGREP_APP_URL="https://semgrep.dev"
+            - export SEMGREP_BASELINE_REF="origin/main"
+            - git fetch origin "+refs/heads/*:refs/remotes/origin/*"
+            - export BITBUCKET_TOKEN=$PAT
+            - semgrep ci
+```
```

Comment:
So this configuration would only work if they are on Bitbucket Pipelines which is one of the many CI providers that can be used with Bitbucket Cloud, which is the SCM. Bitbucket is a SCM that allows for multiple CI providers to be used with it e.g. Jenkins would work as well. 

Again, not sure what's the best way to go about recommending how to export the personal access token (PAT) as BITBUCKET_TOKEN, but maybe we can do something similar to Gitlab where we show one example on how to export the token.

Reply:
I'm honestly not an expert in CI pipelines in general and I don't have access to a testing instance where we could test it out with the other CI providers as well so I'm not quite comfortable recommending blindly on how the CI pipelines should be configured. 

I have a hunch that maybe the CSEs may have repos on Bitbucket that use other CI providers but you may want to double-check. This is also why I decided to generalize step 3 as exporting the personal access token (PAT) as BITBUCKET_TOKEN because in theory, we could do it for any CI provider.

Reply:
As Pang said:

> Also, I'm not sure if we should include this config as the recommended config mainly because it only runs diff scans on pull requests which is not the desired behavior since we also want Semgrep to run full scans on the main branch as well.

This is why we don't mix scanning a project with enabling PR comments. Line 403 gives off a different impression. We can say "here's a sample config" as the header then comment on line 419:

 `- export BITBUCKET_TOKEN=$PAT # This line enables PR comments`

- something like that.

Reply:
Yeah, I'll defer to you both but I do think that giving an example where you just specify that the name of the env var exported has to `BITBUCKET_TOKEN` for PR comments to fire for Bitbucket should be sufficient.

Reply:
@s-santillan can you make concrete suggestions in the PR, please? Trying to figure out what you mean from comments can be hard than comments with direct edits - reading the comment above gives me the impression that: 1. Quote from Pang's comment that states we should include both main branch and topic branch scans in the code snippet. 2. Your comment under it says scanning a project (main branch?) with enabling PR comments. - This seems contradictory. @pangngan you mean to use `BITBUCKET_TOKEN` instead of `PAT`?

Please use the suggestions GitHub feature to speed up this process. 🙏

Reply:
As Sara mentioned previously, the only line that is relevant to enable Bitbucket PR comments to fire is to export the personal access token created under the variable name `BITBUCKET_TOKEN`. In theory, the user can name the workspace/repository varialbe to whatever they want. It just happens to be that in the example I gave in the thread, I just happened to name it `PAT` so for a user using Bitbucket Pipelines, they would have to do something like 
`- export BITBUCKET_TOKEN=$PAT` 
to enable Bitbucket PR comments.

Reply:
Yep I'll try to write something more comprehensive and helpful. Just to manage expectations, I started working on Slack stuff and need some time to context-switch, so expect an update tomorrow. In the meantime @adamkvitek I think you can work on the release notes and data analytics stuff that you started. (FYI I'm done with my section of release notes). I expect the changes to this PR to be minimal (i.e. they are almost there) so don't worry about it.

Reply:
@s-santillan already worked on the release notes the whole day. Feel free to add the items from your side.